src/screens/OtherProfile/hooks/useOtherProfileScreenHooks.tsx

```diff
@@ -116,21 +139,27 @@ const useOtherProfileScreenHooks = (targetUserProfileId: string, username: strin
 
     return () => {
       setOtherProfileFeed([], dispatchOtherProfile);
+      setOtherProfileData(null);
     };
   }, []);
 
   return {
+    username,
     feeds,
     otherProfileData,
     isBlocked,
     isBlocking,
     isLoading,
     isProfileFetching,
+    isCurrentFollowed,
+    otherUserId: other_id,
+    selfUserId: params?.data?.user_id,
 
     refetchOtherProfile: () => fetchOtherProfileData(),
     refetchBlockStatus: () => fetchBlockStatus(false),
 
-    setOtherProfileData
+    setOtherProfileData,
+    setIsCurrentFollowed
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the cleanup function of the `useEffect` hook, you're setting `otherProfileData` to `null`. However, in the initialization of `otherProfileData`, it's set as an empty object. It would be better to maintain consistency by using the same type for initialization and cleanup.

```diff
- setOtherProfileData(null);
+ setOtherProfileData({});
```

Also, the return object of the hook now includes more properties (`username`, `isCurrentFollowed`, `otherUserId`, `selfUserId`, `setIsCurrentFollowed`). This is fine as long as these are necessary and used in the component that uses this hook. If not, consider removing them to keep the interface of the hook clean and minimal.


<!-- This is an auto-generated comment by OSS CodeRabbit -->